code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -770,19 +790,636 @@ nobiliumsuppression = INFINITY
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/turf/open/location = isturf(holder) ? holder : null
+	var produced_amount = min(5, cached_gases[/datum/gas/tritium][MOLES], cached_gases[/datum/gas/proto_nitrate][MOLES])
+	if(cached_gases[/datum/gas/tritium][MOLES] - produced_amount < 0 || cached_gases[/datum/gas/proto_nitrate][MOLES] - produced_amount * 0.01 < 0)
+		return NO_REACTION
+	location.rad_act(produced_amount * 2.4)
+	cached_gases[/datum/gas/tritium][MOLES] -= produced_amount
+	cached_gases[/datum/gas/hydrogen][MOLES] += produced_amount
+	cached_gases[/datum/gas/proto_nitrate][MOLES] -= produced_amount * 0.01
+	energy_released += 50
+	if(energy_released)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+	return REACTING
+
+/datum/gas_reaction/proto_nitrate_hydrogen_response
+	priority = 24
+	name = "Proto Nitrate hydrogen response"
+	id = "proto_nitrate_hydrogen_response"
+
+/datum/gas_reaction/proto_nitrate_hydrogen_response/init_reqs()
+	min_requirements = list(
+		/datum/gas/proto_nitrate = MINIMUM_MOLE_COUNT,
+		/datum/gas/hydrogen = 150,
+	)
+
+/datum/gas_reaction/proto_nitrate_hydrogen_response/react(datum/gas_mixture/air, datum/holder)
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var produced_amount = min(5, cached_gases[/datum/gas/hydrogen][MOLES], cached_gases[/datum/gas/proto_nitrate][MOLES])
+	if(cached_gases[/datum/gas/hydrogen][MOLES] - produced_amount < 0)
+		return NO_REACTION
+	cached_gases[/datum/gas/hydrogen][MOLES] -= produced_amount
+	cached_gases[/datum/gas/proto_nitrate][MOLES] += produced_amount * 0.5
+	energy_released -= produced_amount * 2500
+	if(energy_released)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+	return REACTING
+
+/datum/gas_reaction/proto_nitrate_cyrion_b_response
+	priority = 25
+	name = "Proto Nitrate Cyrion B response"
+	id = "proto_nitrate_cyrion_b_response"
+
+/datum/gas_reaction/proto_nitrate_cyrion_b_response/init_reqs()
+	min_requirements = list(
+		/datum/gas/proto_nitrate = MINIMUM_MOLE_COUNT,
+		/datum/gas/cyrion_b = MINIMUM_MOLE_COUNT,
+		"TEMP" = FIRE_MINIMUM_TEMPERATURE_TO_EXIST
+	)
+
+/datum/gas_reaction/proto_nitrate_cyrion_b_response/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	var/turf/open/location = isturf(holder) ? holder : null
+	var max_power = min(5, cached_gases[/datum/gas/cyrion_b][MOLES])
+	cached_gases[/datum/gas/cyrion_b][MOLES] = 0
+	explosion(location, max_power * 0.55, max_power * 0.95, max_power * 1.25, max_power, max_power* 3)
```

Comment:
This is the arg list for explosion
`(atom/epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range, adminlog = TRUE, ignorecap = FALSE, flame_range = 0, silent = FALSE, smoke = FALSE)`
or
`(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)`
you're giving 6 arguments, and giving the last one as if it's a scaling number, but it's the var that decides if this should be logged to the admins. What was intended here?

Reply:
oof i wanted to use the max_power *3 as flash_range, but it seems i added one too many, i'll fix it